### PR TITLE
Amazon Textract as document loader

### PR DESCRIPTION
**Pull request by 3coins**
Description: Adding support for [Amazon Textract](https://aws.amazon.com/textract/) as a PDF document loader

### CODE REVIEW

**Comment by eyurtsev** [comment]
@3coins tag me whenever you're ready for re-review

**Comment by 3coins** [comment]
@eyurtsev 
Took care of your suggestions:
1. Moved the imports inside the constructor.
2. Updated the typing hint to `Sequence[int]`, int was easier to convert to the Enum for downstream lib.
3. Added docs for the constructor arguments
4. Specifying an S3 path doesn't download the file now, thanks for suggesting this correction 🚀 
5. Updated the logic in parser for accommodating 4.

About your suggestion to work on a S3 blob generator, that's a great idea, and I can work on it if you create the interface. Let me know if there is anything else I should update here.

**Comment by schadem** [comment]
@3coins: rgdg 

>     2. Updated the typing hint to `Sequence[int]`, int was easier to convert to the Enum for downstream lib.

it would be easier for users to pass in a str instead of an int identifying the features (e. g. "FORMS", "TABLES", "SIGNATURES"

**Comment by schadem** [comment]
chatted with @3coins and we will change the int to str

**Comment by 3coins** [comment]
@schadem 
Thanks for updating the PR. Once you post these changes, plz tag @eyurtsev, so he can do the final review and merge.

**Comment by 3coins** [comment]
@eyurtsev 
Done with all updates. Plz review and let us know if you have any other feedback.

File: libs/langchain/langchain/document_loaders/pdf.py [hunk 1]
```diff
@@ -440,3 +456,144 @@ def load(self) -> List[Document]:
...
+    def lazy_load(
+        self,
+    ) -> Iterator[Document]:
+        """Lazy load documents"""
+        # the self.file_path is local, but the blob has to include
+        # the S3 location if the file originated from S3 for multi-page documents
+        # raises ValueError when multi-page and not on S3"""
+
+        if self.web_path and self._is_s3_url(self.web_path):
+            blob = Blob(path=self.web_path)
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/pdf.py:563]: Curious any reason why the Blob is initalized differently here from the line below?

Do you have any advice about Langchain's blob object design? Should we clarify whether it is meant to handle remote paths like s3?

File: libs/langchain/langchain/document_loaders/parsers/pdf.py [hunk 2]
```diff
@@ -149,3 +150,97 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
...
+            )
+        else:
+            textract_response_json = self.tc.call_textract(
+                input_document=blob.as_bytes(),
+                features=self.textract_features,
+                call_mode=self.tc.Textract_Call_Mode.FORCE_SYNC,
+                boto3_textract_client=self.boto3_textract_client,
+            )
+
+        current_text = ""
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/parsers/pdf.py:230]: (not required for merging pr) -- for devs not familiar with the it's unclear what type of information is returned and whether this logic can be assumed to be the "correct" way to project the raw response onto documents

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
@3coins looks good to me, going to merge.

File: libs/langchain/langchain/document_loaders/pdf.py (hunk 1 above)
Comment (3coins) [reply libs/langchain/langchain/document_loaders/pdf.py:563]: The main reason to handle it differently was that for s3 and http paths, the path reference is in web_path, vs for local files. I am not sure why that distinction is needed in the base loader.

File: libs/langchain/langchain/document_loaders/parsers/pdf.py (hunk 2 above)
Comment (3coins) [reply libs/langchain/langchain/document_loaders/parsers/pdf.py:230]: @schadem 
Can you provide explanation for this part? We should add some comments around this in a follow up PR.

**Comment by baskaryan** [comment]
this is super cool! we should add to the docs so folks can find it

File: libs/langchain/langchain/document_loaders/parsers/pdf.py (hunk 2 above)
Comment (schadem) [reply libs/langchain/langchain/document_loaders/parsers/pdf.py:230]: @3coins : Will do. I'll create a follow up PR and also add it to the comments, like @baskaryan mentions below.

**Pull request merged**